File: performance/src/main/java/tech/okcredit/performance/Traces.java
package tech.okcredit.performance;

public final class Traces {


    /**
     * Names for custom traces and custom metrics must meet the following requirements:
     * 1. no leading or trailing whitespace
     * 2. no leading underscore
     * 3. max length is 32 characters
     **/


    /****************************************************************
     Account
     ****************************************************************/
    // Customer And Supplier
    public static final String RENDER_ADD_CUSTOMER = "RenderAddCustomer";
    public static final String RENDER_ADD_SUPPLIER = "RenderAddSupplier";
    public static final String RENDER_ADD_SUPPLIER_Transaction = "RenderAddSupplierTransaction";
    public static final String RENDER_ADD_Transaction = "RenderAddTransaction";
    public static final String RENDER_CUSTOMER = "RenderCustomer";
    public static final String RENDER_DUE_CUSTOMER = "RenderDueCustomer";
    public static final String RENDER_SUPPLIER = "RenderSupplier";
    public static final String RENDER_SUPPLIER_TRANSACTION = "RenderSupplierTransaction";
    public static final String RENDER_SUPPLIER_TUTORIAL = "RenderSupplierTutorial";
    public static final String RENDER_TRANSACTION_DETAILS = "RenderTransactionDetails";
    public static final String OnCreateCustomerProfile = "OnCreateCustomerProfile";

    //Merchant
    public static final String RENDER_MERCHANT_DESTINATION = "RenderMerchantDestination";
    public static final String RENDER_MERCHANT_PROFILE = "RenderMerchantProfile";
    public static final String RENDER_NUMBER_CHANGE = "RenderNumberChange";
    public static final String RENDER_BUSINESS_NAME = "RenderBusinessName";


    //MISC
    public static final String OnCreate_Account = "OnCreateAccount";
    public static final String RENDER_ACCOUNT_STATEMENT = "RenderAccountStatement";
    public static final String RENDER_KNOW_MORE = "RenderKnowMore";


    // Rewards
    public static final String RENDER_REWARDS = "RenderRewards";
    public static final String RENDER_FIREWORK_LOTTIE_AIMATION = "RenderFireWorkLottieAnimation";


    /****************************************************************
     OnBoarding
     ****************************************************************/
    public static final String TRACE_LAUNCHER_AUTH = "TracerLauncherAuth";
    public static final String RENDER_ENTER_OTP = "RenderEnterOtp";
    public static final String RENDER_CONFIRM_PHONE_CHANGE = "RenderConfirmPhoneChange";
    public static final String RENDER_ENTER_MOBILE = "RenderEnterMobile";
    public static final String RENDER_ENTER_NEW_NUMBER = "RenderEnterNewNumber";
    public static final String RENDER_LOGIN_SUCCESS = "RenderLoginSuccess";
    public static final String RENDER_LANGUAGE = "RenderLanguage";
    public static final String RENDER_OTP_VERIFICATION = "RenderOTPVerification";
    public static final String RENDER_ONBOARDING_TUTORIAL = "RenderOnBoardingTutorial";


    /****************************************************************
     Security
     ****************************************************************/
    public static final String RENDER_APPLOCK = "RenderAppLock";
    public static final String RENDER_PAYMENT_PASSWORD = "RenderPaymentPassword";
    public static final String RENDER_PRIVACY = "RenderPrivacy";
    public static final String OnCreate_AppLock = "OnCreateAppLock";


    /****************************************************************
     Platform and Global
     ****************************************************************/

    public static final String OnCreate_Application = "OnCreateApplication";
    public static final String RENDER_SYNC_SCREEN = "RenderSync";
    public static final String onCreateCameraActivity = "onCreateCameraActivity";
    public static final String OnCreate_AppLockPrompt = "OnCreateApplockPrompt";
    public static final String OnCreate_AppLockPref = "OnCreateAppLockPref";

}

File: base/src/main/java/tech/okcredit/base/Traces.kt
package tech.okcredit.base

@Deprecated(message = "Keep constants in respective modules")
object Traces {

    // Customer And Supplier
    const val RENDER_ADD_CUSTOMER = "RenderAddCustomer"
    const val RENDER_ADD_SUPPLIER = "RenderAddSupplier"
    const val RENDER_ADD_SUPPLIER_Transaction = "RenderAddSupplierTransaction"
    const val RENDER_ADD_Transaction = "RenderAddTransaction"
    const val RENDER_CUSTOMER = "RenderCustomer"
    const val RENDER_DUE_CUSTOMER = "RenderDueCustomer"
    const val RENDER_SUPPLIER = "RenderSupplier"
    const val RENDER_SUPPLIER_TRANSACTION = "RenderSupplierTransaction"
    const val RENDER_TRANSACTION_DETAILS = "RenderTransactionDetails"
    const val OnCreateCustomerProfile = "OnCreateCustomerProfile"

    // Merchant
    const val RENDER_MERCHANT_DESTINATION = "RenderMerchantDestination"
    const val RENDER_MERCHANT_PROFILE = "RenderMerchantProfile"
    const val RENDER_NUMBER_CHANGE = "RenderNumberChange"
    const val RENDER_BUSINESS_NAME = "RenderBusinessName"

    // MISC
    const val RENDER_ACCOUNT_STATEMENT = "RenderAccountStatement"
    const val RENDER_KNOW_MORE = "RenderKnowMore"

    /****************************************************************
     * OnBoarding
     */
    const val RENDER_ENTER_OTP = "RenderEnterOtp"
    const val RENDER_CONFIRM_PHONE_CHANGE = "RenderConfirmPhoneChange"
    const val RENDER_ENTER_MOBILE = "RenderEnterMobile"
    const val RENDER_ENTER_NEW_NUMBER = "RenderEnterNewNumber"
    const val RENDER_LANGUAGE = "RenderLanguage"
    const val RENDER_OTP_VERIFICATION = "RenderOTPVerification"
    const val RENDER_ONBOARDING_TUTORIAL = "RenderOnBoardingTutorial"

    /****************************************************************
     * Security
     */
    const val RENDER_APPLOCK = "RenderAppLock"
    const val RENDER_PAYMENT_PASSWORD = "RenderPaymentPassword"
    const val RENDER_PRIVACY = "RenderPrivacy"
    const val OnCreate_AppLock = "OnCreateAppLock"

    /****************************************************************
     * Platform and Global
     */
    const val RENDER_SYNC_SCREEN = "RenderSync"
    const val onCreateCameraActivity = "onCreateCameraActivity"
    const val OnCreate_AppLockPrompt = "OnCreateApplockPrompt"
    const val OnCreate_AppLockPref = "OnCreateAppLockPref"
    const val Trace_syncTxsPerformance = "syncTxsPerformance" // Change Name and Structure of event
    const val Trace_CoreModule_SyncTxsPerformance = "CoreModuleSyncTxsPerformance"
    const val Trace_GetActiveCustomerList = "GetActiveCustomerList" // Change Name and Structure of
    const val Trace_GetActiveCustomerListFromCoreModule = "GetActiveCustomerListFromCoreModule" // event

    const val Trace_BillModule_SyncTxsPerformance = "BillModuleSyncTxsPerformance"
}
